packages/web/src/pages/collectibles-playlist-page/CollectiblesPlaylistPageProvider.tsx

```diff
@@ -91,29 +104,65 @@ export const CollectiblesPlaylistPageProvider = ({
             .sort((a, b) => order.indexOf(a.id) - order.indexOf(b.id))
             // Filter to audio nfts
             .filter(c =>
-              ['mp3', 'wav', 'oga'].some(
+              ['mp3', 'wav', 'oga', 'mp4'].some(
                 ext => c.hasAudio || c.animationUrl?.endsWith(ext)
               )
             )
             // Calculate duration
             .map(async collectible => {
-              const a = new Audio()
-              // Only fetch the metadata
-              a.preload = 'metadata'
-              a.src = collectible.animationUrl
-              const duration = await new Promise(resolve => {
-                a.onloadedmetadata = () => {
-                  a.pause()
-                  resolve(a.duration)
+              if (!collectible.animationUrl) return collectible
+
+              if (
+                ['mp3', 'wav', 'oga'].some(ext =>
+                  collectible.animationUrl?.endsWith(ext)
+                )
+              ) {
+                const a = new Audio()
+                // Only fetch the metadata
+                a.preload = 'metadata'
+                a.src = collectible.animationUrl
+                const duration: number = await new Promise(resolve => {
+                  a.onloadedmetadata = () => {
+                    resolve(a.duration)
+                  }
+                })
+                collectible.duration = duration
+              }
+
+              if (
+                ['mp4'].some(ext => collectible.animationUrl?.endsWith(ext)) &&
```

Comment:
are we potentially expecting more video extension to check in the future? otherwise can just directly check equality with `===`

Reply:
that's what i was thinking, but i've honestly got no clue. you'd know better than me :)

Reply:
also, the 'if' could be an 'else if' to save some nanoseconds :)